lib/web_of_science/process_records.rb

```diff
@@ -89,21 +88,47 @@ def filter_by_identifiers(records)
       end
 
       ## 4
-      # @param [WebOfScience::Record] rec
+      # @param [WebOfScience::Record] record
       # @return [String, nil] WosUID for a new Publication
-      def create_publication(rec)
-        return nil if WebOfScienceSourceRecord.find_by(uid: rec.uid).nil?
-        # All of this is TBD, it might live here or in another class
-        # TODO: For WOS-record that has a PMID, fetch data from PubMed
-        # TODO: create a new Publication, including Publication.pub_hash data
-        # TODO: create new PublicationIdentifiers from rec.identifiers
-        # TODO: associate Publication with Author and create Contribution
-        rec.uid
-      rescue
-        logger.error("#{rec.uid} failed to create Publication")
+      def create_publication(record)
+        pub = Publication.new(active: true, pub_hash: record.pub_hash, xml: record.to_xml)
+        pubmed_additions(record, pub) if record.pmid
+        create_contribution(pub)
+        pub.sync_publication_hash_and_db # creates new PublicationIdentifiers
+        pub.save
+        record.uid
+      rescue StandardError => e
+        NotificationManager.error(e, "#{record.uid} failed to create Publication", self)
         nil
       end
 
+      # For WOS-record that has a PMID, fetch data from PubMed and enhance the pub.pub_hash with PubMed data
+      # @param [WebOfScience::Record] record
+      # @param [Publication] pub
+      # @return [Publication, nil]
```

Comment:
I agree with the TODO this seems like it might be more part of the publication domain, since it performs 2 updates to `pub`.  Return value specification seems not to matter since it is unused.  I don't see how it could ever return `nil` though (it raises instead).

Reply:
- Actually, I _think_ I did that comparison and decided not to use it, because it doesn't do a reverse merge and that's what we need here to preserve the WOS record data and just update or add new data.  But, I'll take a second look at it just to be sure.
- Yea, maybe the return documentation is old, needs updating.